src/shogun/clustering/KMeansBase.cpp

```diff
@@ -130,6 +130,44 @@ void KMeansBase::compute_cluster_variances()
 	}
 }
 
+SGMatrix<float64_t> KMeansBase::compute_std_dev() const
+{
+	require(cluster_centers.size() > 0, "KMeans is not trained!");
+
+	SGMatrix<float64_t> points = distance->get_rhs()
+	                                 ->as<DenseFeatures<float64_t>>()
+	                                 ->get_feature_matrix();
+	SGVector<float64_t> cluster_assignments = const_cast<KMeansBase*>(this)
+	                                              ->apply()
+	                                              ->as<MulticlassLabels>()
+	                                              ->get_labels();
+
+	SGVector<int32_t> counts(k);
+	SGMatrix<float64_t> means = cluster_centers.clone();
+	SGMatrix<float64_t> squares_sums(dimensions, k);
+
+	for (int32_t point_number : range(cluster_assignments.vlen))
+	{
+		auto cluster_number = (int32_t) cluster_assignments[point_number];
+		auto point = points.get_column(point_number);
+		auto& count = counts[cluster_number];
+		auto mean = means.get_column(cluster_number);
+		auto squares_sum = squares_sums.get_column(cluster_number);
+
+		count += 1;
+		auto delta1 = linalg::add(point, mean, 1., -1.);
+		linalg::add(mean, linalg::scale(delta1, 1. / count), mean);
+		auto delta2 = linalg::add(point, mean, 1., -1.);
+		linalg::add(
+		    squares_sum, linalg::element_prod(delta1, delta2), squares_sum);
+	}
+
+	linalg::scale(squares_sums, squares_sums, 1. / (points.num_cols - 1));
+	for (float64_t& x : squares_sums)
+		x = std::sqrt(x);
```

Comment:
is there no linalg for this? Maybe rewrite as std::transform, to be more idiomatic

Reply:
++ for linalg... could send a separate pr for adding an elementwise sqrt. We had a few elementwise operations added recently, you could check those

Reply:
Not sure if there is power function and could just have exponent 0.5?

Reply:
https://github.com/shogun-toolbox/shogun/pull/5016#partial-pull-merging

Reply:
thanks for that other PR